### DISCUSSION TITLE
No pyproject.toml found with alternative mixed Rust/Python project structure

### DISCUSSION THREAD

**Discussion by KCrux** [messense/maturin-action#254] [Q&A]
Hello everyone,
I have a problem when I try to build a wheel when I use the alternative mixed Rust/Python project structure as suggested in the docs (https://www.maturin.rs/project_layout#alternate-python-source-directory-src-layout):
_No pyproject.toml found at /home/runner/work/rainflow_counting/rainflow_counting/rust/pyproject.toml, fallback to latest_

Is there any way to specify the path, as I can to it for the _Cargo.toml_ file with --manifest-path?

If I run on ubuntu-latest, this results in an error. I don't know what exactly happens when I run on windows/macos.

Here are the relevant logs and the ci.yaml (renamed to ci.txt, because Github didn't let me upload it otherwise):
[8_build_windows (windows-latest, x86, 3.10).txt](https://github.com/PyO3/maturin-action/files/14727193/8_build_windows.windows-latest.x86.3.10.txt)
[0_build_linux (ubuntu-latest, x86_64, 3.10).txt](https://github.com/PyO3/maturin-action/files/14727194/0_build_linux.ubuntu-latest.x86_64.3.10.txt)
[ci.txt](https://github.com/PyO3/maturin-action/files/14727195/ci.txt)

Any help is very much appreciated.

**Comment by messense** [comment]
You can pass `--manifest-path` in `args`, or change cwd with `working-directory` option.

Please refer to documentation for detail: https://github.com/PyO3/maturin-action?tab=readme-ov-file#inputs

**Comment by dimaqq** [comment]
I'm also seeing this warning, but I think it's even more confusing in my case:
```
No pyproject.toml found at /home/runner/work/otlp-proto/otlp-proto/pyproject.toml, fallback to latest
```

Here's my project structure:
```
otlp-proto  # git root, what gets checked out
  .git
  pyproject.toml
  otlp_proto.pyi
  Cargo.toml
  build.rs
  src
    lib.rs
```

Why would the action try to find pyproject.toml under doubled repo name?
